examples/hosted-app-action-templates/javascript/build-actions.js

```diff
@@ -1,34 +1,75 @@
-const fs = require('fs/promises');
-const { readdir, readFile, writeFile, mkdir } = fs;
+const esbuild = require('esbuild');
+const { resolve } = require('path');
+const yargs = require('yargs/yargs');
+const { hideBin } = require('yargs/helpers');
 
-const buildActions = async () => {
-  try {
-    console.log('Building app actions');
-    await mkdir('build/actions', { recursive: true });
+const manifest = require('./contentful-app-manifest.json');
+
+const argv = yargs(hideBin(process.argv)).argv;
 
-    const actionsFiles = await readdir('actions', {
-      encoding: 'utf-8',
+const validateActions = () => {
+  const requiredProperties = ['id', 'path', 'entryFile'];
+  const uniqueValues = new Set();
+
+  manifest.actions.forEach((action) => {
+    requiredProperties.forEach((property) => {
+      if (!action.hasOwnProperty(property)) {
+        throw new Error(`Action with name: '${action.name}' is missing the '${property}' property`);
+      }
     });
 
-    const actions = await Promise.all(
-      actionsFiles.map((fileName) => {
-        return readFile(`actions/${fileName}`, {
-          encoding: 'utf-8',
-        });
-      })
-    );
-
-    await Promise.all(
-      actionsFiles.map((fileName, index) => {
-        return writeFile(`build/actions/${fileName}`, actions[index]);
-      })
-    );
-
-    console.log('App actions successfully built');
+    if (uniqueValues.has(action.id)) {
+      throw new Error(`Duplicate action id: '${action.id}'`);
+    }
+    if (uniqueValues.has(action.path)) {
+      throw new Error(`Duplicate action path: '${action.path}'`);
+    }
+    if (uniqueValues.has(action.entryFile)) {
+      throw new Error(`Duplicate entryFile path: '${action.entryFile}'`);
+    }
+
+    uniqueValues.add(action.entryFile);
+    uniqueValues.add(action.path);
+    uniqueValues.add(action.id);
```

Comment:
Very nit; if you're referring to props of an object more than once, I'd destruct them at the top 
( no need to change it though, only if you are really inclined to do so :D )